src/codeu/chat/client/commandline/Chat.java

```diff
@@ -430,12 +494,21 @@ public void invoke(List<String> args) {
           }
         
           if (message.length() > 0) {
-            conversation.add(message);
+          	MessageContext m = conversation.add(message);
           } else {
             System.out.println("ERROR: Messages must contain text");
           }
         }
       }
+      
+      private User findUser(Uuid author) {
+        for (final UserContext user : context.allUsers()) {
```

Comment:
Something to keep in mind for a future optimization (but not necessary for this PR): This runs in O(n) time, but if a different data structure was used to store users, it could run in O(1).